### PR TITLE
Save pass@k result & use custom tokenizer

### PR DESCRIPTION
**Pull request by marianna13**
Hey there!

Two changes that would be nice to have:

1. Save pass@k result after evaluation (in a JSON file).
2. The ability to load custom tokenizer with HF AutoTokenizer (i.e. if tokenizer name is different from the model name).

### CODE REVIEW

File: bigcodebench/evaluate.py [hunk 1]
```diff
@@ -277,6 +277,12 @@ def stucking_checker():
     if not os.path.isfile(result_path):
         with open(result_path, "w") as f:
             json.dump(results, f, indent=2)
+
```
Comment (terryyz) [inline bigcodebench/evaluate.py:280]: Maybe adding `if not os.path.isfile(pass_at_k_path):`?

**Comment by terryyz** [comment]
@marianna13 LGTM, only minor updates are needed. Did you test w/ these changes with some models?

File: bigcodebench/evaluate.py (hunk 1 above)
Comment (terryyz) [reply bigcodebench/evaluate.py:280]: Actually, I think a better way is to check if at least the Pass@1 scores are the same and decide whether we need to rewrite the result_path and pass_at_k_path. Wdyt?

Comment (marianna13) [reply bigcodebench/evaluate.py:280]: yes, sounds good

**Comment by marianna13** [comment]
I tried with `ibm-granite/granite-3b-code-base` and `openai-community/gpt2` tokenizer. Also I tried `apple/OpenELM-1_1B` and `meta-llama/Llama-2-7b-hf` tokenizer ( it didn't work but it's not bc of the tokenizer, there's a problem with OpenELM config)

File: bigcodebench/evaluate.py (hunk 1 above)
Comment (terryyz) [reply bigcodebench/evaluate.py:280]: Thanks! I'll merge the PR after your update.

Comment (marianna13) [reply bigcodebench/evaluate.py:280]: Done

**Review by terryyz — APPROVED** [review]

**Pull request merged**